tools/ports/__init__.py

```diff
@@ -386,6 +403,8 @@ def resolve_dependencies(port_set, settings):
   def add_deps(node):
     node.process_dependencies(settings)
     for d in node.deps:
+      if d not in ports_by_name:
+        utils.exit_with_error(f'Unknown dependency {d} for port {node.name}')
```

Comment:
I actually did find an issue with the way `validate_ports` check for dependencies: it only checks for static dependencies (the `deps` array of a port). As you know from other (built-in) ports, dependencies can be added dynamically (for example, `sdl2_image` adds the `libpng` dependency in the `process_dependencies` call) and if there was a typo there, or adding a dependency that does not exist, then it is not caught via `validate_ports` but it would be caught here. I can remove the check in `validate_ports` then...

Reply:
I did remove the overlap